### DISCUSSION TITLE
Two of the exact same commits deployed, one crashes, the other doesn't

### DISCUSSION THREAD

**Discussion by astonfuture** [vercel/community#4604] [Help] [answered]
### Summary

Running a Remix app on Vercel.

I deployed a commit to production after upgrading some packages. It built fine but the serverless function swere crashing when you actually visit them. I used instant rollback to roll back my production deployment to a previous commit that was working. All good there.

Then I ran git checkout to pull up the old commit that was working. Created a new branch in place (essentially an exact copy of the 
working commit) and made a commit to push that to a preview deployment. The preview deployment is exhbiting the same crashing behaviour in the serverless functions.

My environment variables are all the exact same between production and preview. If I diff the working commit (on production) and the new commit I just made (on preview) there are no differences in the code yet one crashes and the other doesn't. What gives?

### Steps to Reproduce

Would not really be possible to reproduce this locally.

**Comment by astonfuture** [comment]
For some context, the actual crashing error is to do with Remix's streaming. I'm getting the error:

`TypeError: First parameter has member 'readable' that is not a ReadableStream.`

Which I think has something to do with the entry.server.ts but that obviously has not changed so I don't know why it would start exhibiting this behaviour all of a sudden.

**Reply by johny** [reply]
Looks like some internals changed in the runtime, I'm experiencing the same problem on feature branch previews:
```
Unhandled Promise Rejection
{
  "errorType": "Runtime.UnhandledPromiseRejection",
  "errorMessage": "TypeError: First parameter has member 'readable' that is not a ReadableStream.",
  "reason": {
  "errorType": "TypeError",
      "errorMessage": "First parameter has member 'readable' that is not a ReadableStream.",
      "stack": [
        "TypeError: First parameter has member 'readable' that is not a ReadableStream.",
        "    at assertReadableStream (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:362:19)",
        "    at convertReadableWritablePair (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:3524:9)",
        "    at ReadableStream.pipeThrough (/var/task/node_modules/web-streams-polyfill/dist/ponyfill.js:3608:29)", "    at fetchFinale (node:internal/deps/undici/undici:11082:56)", "    at mainFetch (node:internal/deps/undici/undici:10974:9)",
        "    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
      ]
  },
  "promise": { },
  "stack": [
    "Runtime.UnhandledPromiseRejection: TypeError: First parameter has member 'readable' that is not a ReadableStream.",
    "    at process.<anonymous> (file:///var/runtime/index.mjs:1276:17)",
    "    at process.emit (node:events:529:35)",
    "    at process.emit (/var/task/node_modules/source-map-support/source-map-support.js:516:21)", "    at emit (node:internal/process/promises:149:20)",
    "    at processPromiseRejections (node:internal/process/promises:283:27)",
    "    at processTicksAndRejections (node:internal/process/task_queues:96:32)"
  ]
}
Unknown application error occurred
Runtime.Unknown
```

**Reply by matt416** [reply]
I'm experiencing the same issue, suddenly. I thought it was related to today's downtime/issues, but those indicate they're resolved, but this issue isn't.

**Reply by astonfuture** [reply]
I managed to find a (bad, but working) solution, at least for this instance:

I upgraded to a Team's account and transferred the project to that Team. Then I deployed a preview branch which was a copy of an earlier commit and it seemed to work. Perhaps the transfer of the project reset some internal state.

**Reply by visormatt** [reply]
Also running into this issue and opened a ticket as well.

**Reply by astonfuture** [reply]
Are your projects Remix as well? Are you using Supabase alongside them? 

This happened to me specifically after updating the supabase-js and @supabase/auth-helpers-remix packages so I'm trying to work out if those updates were related to the issue or not.

**Reply by visormatt** [reply]
@astonfuture our apps are Remix but not Supabase (not this project at least)

**Comment by emma-sg** [comment]
I'm not sure what the root cause is, but I have a [minimal reproduction](https://github.com/emma-sg/remix-boilerplate), alongside the [corresponding Vercel deployment](https://remix-boilerplate-kubj4g0ko-parentcompanyc.vercel.app/), with logs and source visible. It only seems to occur when a `fetch` call is made from a page's `loader`.

**Reply by visormatt** [reply]
@emma-sg Not sure if the repo is meant to be private or not, but thanks for doing it!

**Reply by emma-sg** [reply]
@visormatt oh whoops, should be public now, my bad!

**Comment by TooTallNate** [comment, marked as answer]
Sorry about that folks. This should be fixed now. You will need to re-deploy your project.

**Reply by visormatt** [reply]
Thanks @TooTallNate and the Vercel team 👏 a re-deploy did the trick for us 💯

**Reply by matt416** [reply]
Redeploy did it over here too, thanks for the fix!

**Reply by johny** [reply]
Thanks!

**Reply by Gbuomprisco** [reply]
@TooTallNate just a question as a result of this: could Vercel work on the way to prevent this from happening, like automated tests?

I received reports this started happening one day before, which is rather alarming since no one had noticed until this issue was opened. I cannot recommend Remix for deployments to my customers with this sort of outage happening and not being monitored for over a day.

Please don't take this is a rant - I only want to understand if there's more work going into monitoring into testing Remix deployments.
